src/Adapters/Storage/Mongo/MongoStorageAdapter.js

```diff
@@ -92,6 +92,10 @@ export class MongoStorageAdapter {
     this._uri = uri;
     this._collectionPrefix = collectionPrefix;
     this._mongoOptions = mongoOptions;
+
+    // MaxTimeMS is not a global MongoDB client option, it is applied per operation.
+    this._maxTimeMS = mongoOptions.maxTimeMS;
+    delete mongoOptions.maxTimeMS;
```

Comment:
Would this delete the property from the options being passed in? Not sure if we clone before passing in - perhaps we should shallow clone here if not?